src/runtime/google-analytics-event-listener.js

```diff
@@ -65,16 +75,61 @@ export class GoogleAnalyticsEventListener {
     }
 
     // TODO(b/234825847): Remove it once universal analytics is deprecated in 2023.
-    if (typeof this.win_.ga === 'function') {
-      this.win_.ga('send', 'event', gaEvent);
+    const ga = this.win_.ga || null;
+    if (isFunction(ga)) {
+      ga(
+        'send',
+        'event',
+        Object.assign(
+          {},
+          gaEvent,
+          !!analyticsParams?.eventCategory
+            ? {
+                eventCategory: analyticsParams.eventCategory,
+              }
+            : null,
+          !!analyticsParams?.eventLabel
+            ? {eventLabel: analyticsParams.eventLabel}
+            : null
+        )
+      );
     }
 
-    if (typeof this.win_.gtag === 'function') {
-      this.win_.gtag('event', gaEvent.eventAction, {
-        'event_category': gaEvent.eventCategory,
-        'event_label': gaEvent.eventLabel,
-        'non_interaction': gaEvent.nonInteraction,
-      });
+    const gtag = this.win_.gtag || null;
+    if (isFunction(gtag)) {
+      gtag(
+        'event',
+        gaEvent.eventAction,
+        Object.assign(
+          {},
+          {
+            'event_category': gaEvent.eventCategory,
+            'event_label': gaEvent.eventLabel,
+            'non_interaction': gaEvent.nonInteraction,
+          },
+          !!analyticsParams?.eventCategory
+            ? {
+                'event_category': analyticsParams.eventCategory,
+              }
+            : null,
+          !!analyticsParams?.surveyQuestion
+            ? {
+                'survey_question': analyticsParams.surveyQuestion || '',
```

Comment:
Needed to add another null check here to satisfy the compiler, but this should never be `null` nor `undefined` after the check on 115, same for `surveyAnswerCategory`.

Reply:
new changes make this comment obsolete.